bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/models/v1/productteaser/ProductTeaserImpl.java

```diff
@@ -173,6 +174,22 @@ public String getCallToActionText() {
         return ctaText;
     }
 
+    @Override
+    public String getCallToActionCommand() {
```

Comment:
Instead of adding a new public method, I would do the check in the `getCallToAction` method and return a new value like `add-to-cart-redirect` which can then indicate for HTL to render an "add to cart" button which however links to the product detail page.

Reply:
I thought of this but it would add more duplication to the markup, so I looked for other options.